test/src/main/java/ai/timefold/solver/test/api/score/stream/SingleConstraintAssertion.java

```diff
@@ -39,36 +98,64 @@ default void penalizesBy(int matchWeightTotal) {
      * @throws AssertionError when the expected penalty is not observed
      */
     default void penalizesBy(long matchWeightTotal) {
-        penalizesBy(matchWeightTotal, null);
+        penalizesBy(null, matchWeightTotal);
     }
 
     /**
-     * As defined by {@link #penalizesBy(int)}.
+     * As defined by {@link #penalizesBy(long)}.
      *
      * @param matchWeightTotal at least 0, expected sum of match weights of matches of the constraint.
      * @param message sometimes null, description of the scenario being asserted
      * @throws AssertionError when the expected penalty is not observed
+     *
+     * @deprecated Use {@link #penalizesBy(String, long)} instead
      */
-    void penalizesBy(long matchWeightTotal, String message);
+    @Deprecated(forRemoval = true, since = "1.7.0")
+    default void penalizesBy(long matchWeightTotal, String message) {
+        penalizesBy(message, matchWeightTotal);
+    }
 
     /**
-     * As defined by {@link #penalizesBy(int)}.
+     * As defined by {@link #penalizesBy(long)}.
+     *
+     * @param message sometimes null, description of the scenario being asserted
+     * @param matchWeightTotal at least 0, expected sum of match weights of matches of the constraint.
+     * @throws AssertionError when the expected penalty is not observed
+     */
+    void penalizesBy(String message, long matchWeightTotal);
+
+    /**
+     * As defined by {@link #penalizesBy(long)}.
      *
      * @param matchWeightTotal at least 0, expected sum of match weights of matches of the constraint.
      * @throws AssertionError when the expected penalty is not observed
      */
     default void penalizesBy(BigDecimal matchWeightTotal) {
-        penalizesBy(matchWeightTotal, null);
+        penalizesBy(null, matchWeightTotal);
     }
 
     /**
-     * As defined by {@link #penalizesBy(int)}.
+     * As defined by {@link #penalizesBy(BigDecimal)}.
      *
      * @param matchWeightTotal at least 0, expected sum of match weights of matches of the constraint.
      * @param message sometimes null, description of the scenario being asserted
      * @throws AssertionError when the expected penalty is not observed
+     *
+     * @deprecated Use {@link #penalizesBy(String, BigDecimal)} instead
```

Comment:
If it starts with a capital letter, it is a sentence and therefore it ends with a `.`.